pyproject.toml

```
@@ -1,7 +1,82 @@
 [build-system]
-requires = [ "setuptools", "wheel" ]
+requires = ["setuptools >= 63.0.0"]
 build-backend = "setuptools.build_meta"
 
+[project]
+name = "bitsandbytes"
+dynamic = ["version"]
+description = "k-bit optimizers and matrix multiplication routines."
+authors = [{name="Tim Dettmers", email="dettmers@cs.washington.edu"}]
+requires-python = ">=3.8"
+readme = "README.md"
+license = {file="LICENSE"}
+keywords = [
+    "gpu",
+    "optimizers",
+    "optimization",
+    "8-bit",
+    "quantization",
+    "compression"
+]
+classifiers = [
+    "Development Status :: 4 - Beta",
+    "License :: OSI Approved :: MIT License",
+    "Environment :: GPU :: NVIDIA CUDA :: 11",
+    "Environment :: GPU :: NVIDIA CUDA :: 12",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "Operating System :: POSIX :: Linux",
+    "Operating System :: MacOS",
+    "Operating System :: Microsoft :: Windows",
+    "Programming Language :: C++",
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Topic :: Scientific/Engineering :: Artificial Intelligence"
+]
+dependencies = [
+    "torch>=1.11,!=1.12.0",
+    "numpy>=1.17"
+]
+
+[project.optional-dependencies]
+benchmark = ["pandas", "matplotlib"]
+docs = [
+    "hf-doc-builder @ git+https://github.com/huggingface/doc-builder.git@main"
+]
+dev = [
+    "bitsandbytes[test]",
+    "build>=1.0.0,<2",
+    "ruff~=0.2.0",
+    "pre-commit>=3.5.0,<4",
+    "wheel>=0.42,<1"
+]
+test = [
+    "einops~=0.6.0",
+    "lion-pytorch==0.0.6",
+    "pytest==7.4.4",
+    "pytest-cases>=3.7.0,<4",
+    "pytest-sugar>=1.0.0,<1.1.0",
+    "scipy>=1.10,<2",
+    "transformers>=4.30.1,<5"
```

Cases isn't used, and Sugar is only for nicer CLI output. Pytest doesn't need to be strictly pinned IMO.

```suggestion
    "einops~=0.6.0",
    "lion-pytorch==0.0.6",
    "pytest~=7.4",
    "scipy>=1.10,<2",
    "transformers>=4.30.1,<5"
```